src/charmed_kubeflow_helpers/lightkube/utilities/_kubernetes_resource_handler.py

```diff
@@ -0,0 +1,152 @@
+from ...lightkube.exceptions import ReconcileError
+from ...status_handling import CharmStatusType
+from ..types import LightkubeResourcesList
+
+
+class KubernetesResourceHandler:
+    """Defines an API for handling Kubernetes resources in charm code
+    """
+
+    def __init__(
+        self,
+        template_files_factory: Callable[[], Iterable[str]],
+        context_factory: Callable[[], dict],
+        field_manager: str,
+        logger: Optional[logging.Logger] = None,
+    ):
+        """
+        Returns a KubernetesResourceHandler instance
+
+        Args:
+            template_files_factory (Callable): A callable that accepts no arguments and returns an
+                                               iterable of template files to render
+            context_factory: A callable that requires no arguments returns a dict of context for
+                             rendering the templates
+            field_manager: The name of the field manager to use when using server-side-apply in
+                           kubernetes.  A good option for this is to use the application name
+                           (eg: `self.model.app.name`).
+            logger (logging.Logger): (Optional) A logger to use for logging (so that log messages
+                                     emitted here will appear under the caller's log namespace).
+                                     If not provided, a default logger will be created.
+        """
+        self._template_files_factory = template_files_factory
+        self._context_factory = context_factory
+        self._field_manager = field_manager
+
+        if logger is None:
+            self.log = logging.getLogger(__name__)  # TODO: Give default logger a better name
+        else:
+            self.log = logger
+
+        self._lightkube_client = None
+
+    def status(self, resources: Optional[LightkubeResourcesList] = None) -> CharmStatusType:
+        """Computes the status of the managed resources as defined by the manifest
+
+        The returned status is the worst status encountered, as defined in
+        .lightkube.utilities.get_first_worst_error (blocked is worse than waiting, waiting is
+        worse than active).  When multiple of the worst statuses are encountered, the first is
+        returned.
+
+        TODO: This method will not notice that we have an extra resource (eg: if our
+              render_manifests() previously output some Resource123, but now render_manifests()
+              does not output that resource.
+        TODO: This method directly logs errors to .log.  Is that a problem?  Maybe we should just
+              return those errors?  Or could have a separate function that does that.
+        """
+        self.log.info("Computing status")
+
+        if resources is None:
+            resources = self.render_manifests()
+
+        charm_ok, errors = check_resources(self.lightkube_client, resources)
+        if charm_ok:
+            self.log.info("Status: active")
```

Comment:
```suggestion
            self.log.info("Suggested status: active")
```
Let's just make sure this log does not causes users to think the charm is somehow active